File: vs2003_cd01/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/Samples/Technologies/Remoting/Basic/RemotingEvents/EventFire/Zap.cs
/*=====================================================================

  File:      Zap.cs

---------------------------------------------------------------------
This file is part of the Microsoft .NET Framework SDK Code Samples.

  Copyright (C) Microsoft Corporation.  All rights reserved.

This source code is intended only as a supplement to Microsoft
Development Tools and/or on-line documentation.  See these other
materials for detailed information regarding Microsoft code samples.
 
THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY
KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
PARTICULAR PURPOSE.
=====================================================================*/

using System;
using System.Runtime.Remoting;
using System.Collections;

namespace Zap
{
    // Define the event arguments
    [Serializable]
    public class GreetingEventArgs : EventArgs
    {
        public GreetingEventArgs(string greeting)
        {
            this.greeting = greeting;
        }

        public string greeting;
    }

    // Define the event
    public delegate void GreetingEvent (object sender, GreetingEventArgs e);

    // Define the Service
    public class Waz : MarshalByRefObject
    {
        // The client will subscribe and 
        // unsubscribe to this event
        public event GreetingEvent Greeting;

        // Method called remotely by client
        public void HelloMethod(string greeting)
        {
            Console.WriteLine("Received String {0}", greeting);

            // Package String in GreetingEventArgs
            GreetingEventArgs e = new GreetingEventArgs(greeting);

            // Fire Event
            if (Greeting != null)
            {
                Console.WriteLine("Firing Event");
                Greeting(this, e);
            }
        }
    }
}

File: vs2003_cd01/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/Samples/Technologies/Remoting/VB/Basic/RemotingEvents/EventFire/Zap.vb

Imports System
Imports System.Runtime.Remoting
Imports System.Collections


Namespace Zap
    
    ' Define the event arguments
    <Serializable()> Public Class GreetingEventArgs
        Inherits EventArgs
        
        Public Sub New(greeting As String)
            Me.greeting = greeting
        End Sub 'New
        
        Public greeting As String
    End Class 'GreetingEventArgs
    
    
    ' Define the event
    Public Delegate Sub GreetingEvent(sender As Object, e As GreetingEventArgs)
    
    
    ' Define the Service
    Public Class Waz
        Inherits MarshalByRefObject
        ' The client will subscribe and 
        ' unsubscribe to this event
        Public Event Greeting As GreetingEvent
        
        
        ' Method called remotely by client
        Public Sub HelloMethod(greeting As String)
            Console.WriteLine("Received String {0}", greeting)
            
            ' Package String in GreetingEventArgs
            Dim e As New GreetingEventArgs(greeting)
            
            ' Fire Event
            If Not (Greeting Is Nothing) Then
                Console.WriteLine("Firing Event")
                RaiseEvent Greeting(Me, e)
            End If
        End Sub 'HelloMethod
    End Class 'Waz
End Namespace 'Zap